packages/opentelemetry-core/src/trace/GlobalTracer.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { NoopTracer } from './NoopTracer';
+
+const noopTracer = new NoopTracer();
+let _globalTracer: types.Tracer | null = null;
+
+export class GlobalTracerDelegate implements types.Tracer {
+  getCurrentSpan(): types.Span {
+    const tracer = _globalTracer || noopTracer;
+    // tslint:disable-next-line:no-any
+    return tracer.getCurrentSpan.apply(tracer, arguments as any);
```

Comment:
What happens if you remove the `as any`?

Reply:
`Argument of type 'IArguments' is not assignable to parameter of type <METHOD SIGNATURE HERE>`. Interestingly, OpenTracing compiled fine without the `as any`. I did not look into whether some sort of rule was put in outside of the global tracer file which makes it ignore these types of errors. This seems like a pretty important error, so I doubt it was suppressed over there.

Reply:
If i remember correctly it's a typing issue, `apply` expect an array but `arguments` are array-like.

Reply:
Would it work to just change it from `as any` to `as IArguments`?

Reply:
Actually, it looks like the type for `getCurrentSpan` is that it takes no arguments (see https://github.com/open-telemetry/opentelemetry-js/blob/fdd896dc27b6813896849b944ce4fd83a1a05322/packages/opentelemetry-types/src/trace/tracer.ts#L36)

Given that, could you just do `return tracer.getCurrentSpan()`?

Reply:
I tried `as IArguments` and got the same error message. Maybe something changed with this TS version to make `IArgument` conversions it more strict?
> Given that, could you just do `return tracer.getCurrentSpan()`?
For this, I intentionally had it use `.apply(...)` so that it is compat with any future interface changes and for it to be consistent with how the rest of the delegate methods are called. Also If a user does any sort of patching/middleware, this would be the only method that would not work. Probably not likely, but I think it makes it more resilient/consistent.

Reply:
Looks like we have a discussion below about simply removing the delegate, which makes this function call no longer needed anyway. Let's see if that solves this!